plugin/dapp/x2Ethereum/executor/oracle/keeper.go

```diff
@@ -0,0 +1,252 @@
+package oracle
+
+import (
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+
+	dbm "github.com/33cn/chain33/common/db"
+	log "github.com/33cn/chain33/common/log/log15"
+	types2 "github.com/33cn/chain33/types"
+	"github.com/33cn/plugin/plugin/dapp/x2Ethereum/types"
+)
+
+var (
+	//日志
+	olog = log.New("module", "oracle")
+)
+
+type OracleKeeper struct {
+	db dbm.KV
+	// 通过审核的最低阈值
+	ConsensusThreshold int64
+}
+
+func NewOracleKeeper(db dbm.KV, ConsensusThreshold int64) *OracleKeeper {
+	if ConsensusThreshold <= 0 || ConsensusThreshold > 100 {
+		return nil
+	}
+	return &OracleKeeper{
+		db:                 db,
+		ConsensusThreshold: ConsensusThreshold,
+	}
+}
+
+func (k *OracleKeeper) GetProphecy(id string) (*types.ReceiptEthProphecy, error) {
+	if id == "" {
+		return NewEmptyProphecy(), types.ErrInvalidIdentifier
```

Comment:
这里和后面的出错的NewEmptyProphecy()  可以直接返回Nil， 不需要构造
```suggestion
		NewEmptyProphecyreturn (), types.ErrInvalidIdentifier
```